sympy/solvers/solveset.py

```diff
@@ -428,6 +442,55 @@ def _solve_trig(f, symbol, domain):
         return ConditionSet(symbol, Eq(f_original, 0), S.Reals)
 
 
+def _solve_trig2(f, symbol, domain):
+    """Secondary helper to solve trigonometric equations,
+    called when first helper fails """
+    from sympy import lcm, expand_trig, degree, simplify
+    f = trigsimp(f)
+    f_original = f
+    trig_functions = f.atoms(sin, cos, tan, sec, cot, csc)
+    trig_arguments = [e.args for e in trig_functions]
+    denominators = []
+    for ar in trig_arguments:
+        try:
+            poly_ar = Poly(ar, symbol)
+        except ValueError:
+            raise ValueError("give up, we can't solve if this is not a polynomial in x")
+        if poly_ar.degree() > 1:  # degree >1 still bad
+            raise ValueError("degree of variable inside polynomial should not exceed one")
+        if poly_ar.degree() == 0:  # degree 0, no x, don't care
+            continue
+        c = poly_ar.all_coeffs(x)[0]   # got the coefficient of x
+        denominators.append(Rational(c).q)
```

Comment:
Does it make sense to remove the gcd of the numerators, too?

Reply:
I don't think numerators would matter. We just want to change the fraction coefficient of `x` into an integer. If any of the coefficients of `x` remains fractional, then `expand_trig()` cant convert it into `tan(x)` , for doing so there is no need to do anything with numerator.
eg:  `m = sin(S(3)*x/2) + sin(5*x)`
     replacing `x` with `2*2*s`  will make the coefficients as `6*s` and `20*s` respectively which can be easily converted to respective `tan` with an integer coefficient by `expand_trig()`.  `3` and `5` being there don't gonna affect it because they have nothing to do with coefficient being fractional.
So, I don't think there would be any role of numerator in here : )

Reply:
If you rewrite the expression you suggested in terms of tan and replace the x with `4s` or `8s` (either of which clears the numerator) you end up with a very different set of equations to solve. By not being economical with the replacement you would be forced to solve a larger set of equations -- of higher order:

```
>>> [degree(i.subs(tan(s),y),y) for i in (factor(expand_trig(eq.rewrite(tan).subs(x,8*s)))).as_numer_denom()[0].args]
[0,1,1,6,6,12,12,1]

>>> [degree(i.subs(tan(s),y),y) for i in (factor(expand_trig(eq.rewrite(tan).subs(x,4*s)))).as_numer_denom()[0].args]
[0,6,12,1]
```

Granted, in *this* case, there would be no reason to use 8 instead of 4 but if the original expression were `sin(S(3)*x/2) + sin(6*x)` it would be better to rewrite replacing x with `4s/3` instead of `4s`, wouldn't it?